src/Umbraco.Core/Webhooks/Events/ContentDeleteWebhookEvent.cs

```diff
@@ -19,7 +19,8 @@ public class ContentDeleteWebhookEvent : WebhookEventContentBase<ContentDeletedN
             webHookService,
             webhookSettings,
             serverRoleAccessor,
-            Constants.WebhookEvents.ContentDelete)
+            Constants.WebhookEvents.ContentDelete,
+            WebhookEventType.Content)
```

Comment:
Would it be better to pass in e.g. something like `new WebhookContentEvent(Constants.WebhookEvents.Content.Delete)`, so it isn't possible to set e.g. `Constants.WebhookEvents.ContentDelete, WebhookEventType.Media)`?

Reply:
I'm not opposed to that idea, but I do like that you could do `Constants.WebhookEvents.ContentDelete, WebhookEventType.Media`, it gives the freedom for whoever is creating their own webhook events to choose 🤔
Especially as changing these values only changes the behavior in the front-end 🤔 
Just my 2 cents 😅 🤷

Reply:
Could we have `WebhookContentEvent()`, `WebhookMediaEvent()`, `WebhookMemberEvent()` and maybe `WebhookCustomEvent()`? .. or maybe just a base `WebhookBaseEvent()` the others inherits from? 🤷‍♂️

It may also be easier to add additional properties later.
I wonder if it should be possible to assign a "friendly" name to the webhook created, so one know what it's for?

e.g. it's an option on Bitbucket, but not on Github.
https://magefan.com/blog/add-webhook-in-bitbucket

Maybe @warrenbuckley or @KevinJump have any thoughts on this?

Reply:
I would vote for Webhooks having an alias and a name, so that they can be displayed nicely, and the alias be the thing that is referenced internally? 

i already changed the name of a class and lost the webhooks (its not a big deal, but refactoring code names shouldn't loose settings ideally?).

I think the `WebhookBaseEvent` class  covers most of the extension senarios, i can think off, not sure 

I stopped short of adding an attribute in the PR for the base Event, but i wonder now, if that would make it easier to define. (like trees, healthchecks etc) ?

```cs
[WebhookEvent("myAlias", "Webhook Name",  "myEventType")]
public class myCustomWebhookEvent : WebhookEventBase {
```

Reply:
Are we meaning a friendly name for the Event? Or friendly name for a the webhook itself ? 🤔 

But as I see it, we have two ways of doing this.

Either we do it as it is now:

```
: base(
            webhookFiringService,
            webHookService,
            webhookSettings,
            serverRoleAccessor,
            "MyFriendName"
            "myAlias",
            "myEventType")
```

or as Kevin suggests:

```
[WebhookEvent("myAlias", "MyFriendName", "myEventType")] public class myCustomWebhookEvent : WebhookEventBase {
```


Which do you prefer as implementers of this ? 🤔

Reply:
Yes - I think we have a friendly name for the webhook event?  
I was wondering if we add the Alias to, for internal tracking ?

Reply:
Right now we're tracking by Event name, so we defintely need someting else to track by, as if you're chaning the name of your event, it will register as a whole other one. 
Maybe we should just do a key instead of an alias though 🙈 
So you could now use the "Name" as the friendly name of the event, but we still track by key so you can change it 👍

Reply:
(and then add a friendly name for a webhook)

Reply:
I guess the webhook itself could have a friendly name, but could maybe also be the webhook event?

But mainly that the entries are displayed with a name, e.g. "My Homepage Event".

![image](https://github.com/umbraco/Umbraco-CMS/assets/2919859/3b323de7-7f3d-4138-8f3b-df53a29976bc)

Reply:
But do you want that to differ from the events you're picking? 🤔 
I personally would be confused if I picked a "ContentPublish" event, but in the overview it had a different name 🙈 

I think the happy medium for me is tracking by Key, so you could give the events whatever name you want. Thus a "My homepage event" would still be possible 👍

Reply:
Well yes, I think that's what some other services using webhooks have as an option:
https://magefan.com/blog/add-webhook-in-bitbucket

in some payment gateways like Adyen, they have a name like "Standard webhook", a optional description and then one selected the events in which case it should trigger:
https://docs.adyen.com/development-resources/webhooks/

Reply:
An example of creating a Discord Webhook 😅
https://configu.com/docs/webhook-discord/

![image](https://github.com/umbraco/Umbraco-CMS/assets/2919859/85d8acaa-5768-430f-b7ee-274dfe8ac5a5)

Reply:
Sometimes it seems both a name and description are possible to set.

![image](https://github.com/umbraco/Umbraco-CMS/assets/2919859/4f47b834-f426-4957-a8a3-d08cebd1ee1d)

Reply:
Ah I see what you mean.
I guess it's "just" adding a Title to a given webhook, so it should be relatively simple.
It will have to be a separate PR though! 🤣